File: problems/005/005.f90
PROGRAM F005

    ! Copyright 2021 Melwyn Francis Carlo

    IMPLICIT NONE

    INTEGER :: I, N

    I = 1
    DO
        N = 20 * I
        ! BASED ON THE RULES OF DIVISIBILITY, NOT ALL NUMBERS FROM 1 TO 20 
        ! HAVE TO BE TESTED AGAINST.
        IF  ( (MOD(N, 19) == 0) .AND. (MOD(N, 17) == 0)   & 
        .AND. (MOD(N, 16) == 0) .AND. (MOD(N, 13) == 0)   &
        .AND. (MOD(N, 11) == 0) .AND. (MOD(N, 9)  == 0)   &
        .AND. (MOD(N, 8)  == 0) .AND. (MOD(N, 7)  == 0) ) THEN
            EXIT
        END IF
        I = I + 1;
    END DO

    PRINT ('(I0)'), N

END PROGRAM F005

File: problems/005/005.c
#include <stdio.h>

int main() {
    int i = 1;
    while (1) {
        long int n = 20 * i;
        // Based on the rules of divisibility, not all numbers from 1 to 20 
        // have to be tested against.
        if ( ((n % 19) == 0) && ((n % 17) == 0) && ((n % 16) == 0) 
        && ((n % 13) == 0) && ((n % 11) == 0) && ((n % 9) == 0) 
        && ((n % 8) == 0) && ((n % 7) == 0) ) {
            printf("\n%ld\n\n", n);
            break;
        }
        i += 1;
    }
    return 0;
}
